[cli11] Update to v2.4.0

patrikhuber opened the pull request:
- Updated CLI11 version and hash to v2.4.0
- Ran vcpkg.exe x-add-version cli11
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

patrikhuber committed:
Updated CLI11 version and hash to v2.4.0

patrikhuber committed:
Ran vcpkg.exe x-add-version cli11

patrikhuber commented:
Trying to update CLI11 to v2.4.0, which was released earlier today. It's failing on a couple of platforms, I'm not sure why - I don't know if/how I'd be able to open the Azure pipeline log `D:\b\cli11\install-x64-uwp-dbg-out.log`

I also don't know what these three lines do and if they're still needed - the first two are probably standard, the third one I don't know  whether this is still needed or whether this was "fixed" by the CLI11 update: https://github.com/microsoft/vcpkg/blob/fecdd89f28ed883b26a48c2b30085f6384070bb8/ports/cli11/portfile.cmake#L20-L22

If someone could help getting this over the line, that would be great. @kiwixz CC @henryiii @phlptp

henryiii commented:
As far as I can tell, Microsoft has deprecated UWP a few years ago (https://www.thurrott.com/dev/258377/microsoft-officially-deprecates-uwp), and I'm not sure a CLI interface makes much sense in a UWP app anyway. So I think the best course of action might be to disable UWP?

patrikhuber committed:
Skip CLI11 build on UWP

CLI11's Unicode support doesn't build on UWP. As UWP is deprecated, it most likely won't be supported in the future.

patrikhuber commented:
Thank you both! I've pushed an update that skips the CI build on UWP. Let's see if it'll be all green now.

dg0yt commented on scripts/ci.baseline.txt:
```diff
@@ -187,6 +187,10 @@ lapack-reference:arm64-windows=skip
 lapack-reference:arm64-uwp=skip
 lapack-reference:x64-uwp=skip
 
+# cli11 Unicode support doesn't work on UWP
+cli11:x64-uwp=skip
+cli11:arm64-uwp=skip
+
```
```suggestion
```
Not this...

dg0yt commented on ports/cli11/vcpkg.json:
```diff
@@ -1,6 +1,6 @@
+  "version": "2.4.0",
   "description": "CLI11 is a command line parser for C++11 and beyond that provides a rich feature set with a simple and intuitive interface.",
   "homepage": "https://github.com/CLIUtils/CLI11",
   "license": "BSD-3-Clause",
```
but this:
```suggestion
  "license": "BSD-3-Clause",
  "supports": "!uwp"
```

patrikhuber committed:
Removed cli11 from scripts/ci.baseline.txt

Co-authored-by: Kai Pastor <dg0yt@darc.de>

patrikhuber committed:
Added supports: !uwp to ports/cli11/vcpkg.json

cli11 Unicode support doesn't work on UWP

Co-authored-by: Kai Pastor <dg0yt@darc.de>

patrikhuber committed:
Updated versions/cli11.json

patrikhuber commented:
It might be potentially good to go now?

rapidparty commented:
This update broke our Visual Studio compilation in Azure CICD pipeline. 

##[error]D:\a\1\vcpkg\installed\x86-windows-static\include\CLI\Encoding.hpp(53,10): Error C1083: Cannot open include file: 'impl/Encoding_inl.hpp': No such file or directory

The compilation error happens because the folder "impl" and its .hpp files are not extracted from the CLIUtils-CLI11-v2.4.0.tar.gz file and placed in the vcpkg include folder during the installation of the library with vcpkg.

kiwixz commented:
I had the same problem because one target was not properly linking to CLI11 in CMake.  You should double check if that's also the case for you.

MonicaLiu0311 commented:
I haven't reproduced your issue, could you please provide a reproduction code?
```
1> CMake generation started for configuration: 'x64-Debug'.
1> Command line: "C:\Windows\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\ENTERPRISE\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 17 2022" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\monica\source\repos\CMakeUsage\out\install\x64-Debug"  -DCMAKE_TOOLCHAIN_FILE=G:/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\monica\source\repos\CMakeUsage" 2>&1"
1> Working directory: C:\Users\monica\source\repos\CMakeUsage\out\build\x64-Debug
1> [CMake] -- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
1> [CMake] -- The C compiler identification is MSVC 19.38.33135.0
1> [CMake] -- The CXX compiler identification is MSVC 19.38.33135.0
1> [CMake] -- Detecting C compiler ABI info
1> [CMake] -- Detecting C compiler ABI info - done
1> [CMake] -- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.38.33130/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting C compile features
1> [CMake] -- Detecting C compile features - done
1> [CMake] -- Detecting CXX compiler ABI info
1> [CMake] -- Detecting CXX compiler ABI info - done
1> [CMake] -- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.38.33130/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting CXX compile features
1> [CMake] -- Detecting CXX compile features - done
1> [CMake] -- Configuring done (8.0s)
1> [CMake] -- Generating done (0.0s)
1> [CMake] -- Build files have been written to: C:/Users/monica/source/repos/CMakeUsage/out/build/x64-Debug
1> Extracted CMake variables.
1> Extracted source files and headers.
1> Extracted code model.
1> Extracted toolchain configurations.
1> Extracted includes paths.
1> CMake generation finished.

```

```
>------ Build All started: Project: CMakeUsage, Configuration: x64-Debug ------
  MSBuild version 17.8.5+b5265ef37 for .NET Framework
  
    1>Checking Build System
    Building Custom Rule C:/Users/monica/source/repos/CMakeUsage/CMakeUsage/CMakeLists.txt
    CMakeUsage.cpp
    main.vcxproj -> C:\Users\monica\source\repos\CMakeUsage\out\build\x64-Debug\CMakeUsage\Debug\main.lib
    Building Custom Rule C:/Users/monica/source/repos/CMakeUsage/CMakeLists.txt

Build All succeeded.
```



<details>
<summary><b>test.cpp</b></summary>
<pre>
#include &#60;iostream&#62;
#include "CLI/App.hpp"

using namespace std;

int main()
{
        cout << "Hello CMake." << endl;
        return 0;
}
</pre></details>


<details>
<summary><b>CMakeLists.txt</b></summary>
<pre>
cmake_minimum_required (VERSION 3.8)

set(CMAKE_TOOLCHAIN_FILE "G:/vcpkg/scripts/buildsystems/vcpkg.cmake")

project ("test")

add_library (main "test.cpp")

find_package(CLI11 CONFIG REQUIRED)
target_link_libraries(main PRIVATE CLI11::CLI11)
</pre></details>

rapidparty commented:
> I had the same problem because one target was not properly linking to CLI11 in CMake. You should double check if that's also the case for you.

Sorry, I suck at CMAKE. How did you end up resolving this in your build environment.

kiwixz commented:
You need to add this:
```cmake
find_package(CLI11 REQUIRED)
target_link_libraries(the_target_that_doesnt_build_anymore PUBLIC CLI11::CLI11)
```
